pydatastructs/miscellaneous_data_structures/stack.py

```diff
@@ -81,47 +82,35 @@ def peek(self):
         return None
 
 class ArrayStack(Stack):
+    __slots__ = ['size', 'items', 'dtype']
 
-    __slots__ = ['maxsize', 'top', 'items', 'dtype']
-
-    def __new__(cls, maxsize=None, top=0, items=None, dtype=int):
-        if not _check_type(maxsize, int):
-            raise ValueError("maxsize is missing.")
-        if not _check_type(top, int):
-            raise TypeError("top is not of type int.")
+    def __new__(cls, size=None, items=None, dtype=int):
+        if not _check_type(size, int):
+            raise ValueError("size is missing.")
         if items is None:
-            items = OneDimensionalArray(dtype, maxsize)
-        if not _check_type(items, OneDimensionalArray):
+            items = DynamicOneDimensionalArray(dtype, size)
```

Comment:
Do we even need `size` too? We can remove it safely. APIs will break which will require changes in other modules too.

Reply:
We can do, `DynamicOneDimensionalArray(dtype, 0)`.